http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -269,63 +234,15 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
             ctx.writeAndFlush(new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.INTERNAL_SERVER_ERROR));
             return;
         }
-        routeExecutor.filterPublisher(new AtomicReference<>(nettyHttpRequest), () -> routeExecutor.onError(cause, nettyHttpRequest))
-            .onComplete((response, throwable) -> writeResponse(ctx, nettyHttpRequest, response, throwable));
+        new NettyRequestLifecycle(this, ctx, nettyHttpRequest).handleException(cause);
```

Comment:
IMHO the design where `RoutingInBoundHandler` uses `NettyRequestLifecycle` and then it's using `RoutingInBoundHandler` is not the best one.

Reply:
i moved one method (findFile) to NRL. The other call to the RIB is `writeResponse`. it is nice to have a separation between the request and response handling logic. maybe it doesnt make sense though. i am keeping it in RIB for now, if only to not blow up this pr further.

there are also various fields on RIB that are accessed, most come from the app context (eg the io executor, or the route executor). these could be passed in individually into the constructor. however that is not worth it as long as there is a link to RIB anyway for response writing.